### PR TITLE
community/rust: upgrade to 1.19

### PR DESCRIPTION
**Pull request by jeizsm**

### CODE REVIEW

File: community/rust/APKBUILD [hunk 1]
```diff
@@ -54,7 +52,7 @@ prepare() {
 	cd "$builddir"
 
 	# Remove bundled dependencies.
-	rm -Rf src/llvm/ src/jemalloc/
+	rm -Rf src/llvm/
```
Comment (jirutka) [inline community/rust/APKBUILD]: Why did you remove it? There’s still src/jemalloc in the tarball.

File: community/rust/alpine-allow-crt-static.patch [hunk 2]
```diff
@@ -1,41 +0,0 @@
...
-         return crt_static;
---- a/src/test/compile-fail/crt-static-gated.rs
-+++ b/src/test/compile-fail/crt-static-gated.rs
-@@ -10,5 +10,6 @@
- 
- // compile-flags:-C target-feature=+crt-static
- // error-pattern: specifying the `crt-static` target feature is only allowed
-+// ignore-test we just want this feature and no one will stop us!
- 
- fn main() {}
```
Comment (jirutka) [inline community/rust/alpine-allow-crt-static.patch]: Are you sure that this is not needed anymore? They said that this will be always gated (i.e. allowed only in nightly).

File: community/rust/musl-support-dynamic-linking.patch [hunk 3]
```diff
@@ -38,14 +36,21 @@ diff --git a/src/bootstrap/compile.rs b/src/bootstrap/compile.rs
...
 +    if target.contains("musl") {
          copy_musl_third_party_objects(build, target, &libdir);
      }
+ 
+     if build.config.sanitizers && compiler.stage != 0 && target == "x86_64-apple-darwin" {
+         // The sanitizers are only built in stage1 or above, so the dylibs will
+         // be missing in stage0 and causes panic. See the `std()` function above
+         // for reason why the sanitizers are not built in stage0.
+         copy_apple_sanitizer_dylibs(&build.native_dir(target), "osx", &libdir);
+     }
```
Comment (jirutka) [inline community/rust/musl-support-dynamic-linking.patch]: Why is it here? This is relevant only on macOS…?

File: community/rust/APKBUILD (hunk 1 above)
Comment (jeizsm) [reply community/rust/APKBUILD]: It doesn't compile without it despite of --disable-jemalloc, but I'm pretty sure jemalloc is unused

File: community/rust/alpine-allow-crt-static.patch (hunk 2 above)
Comment (jeizsm) [reply community/rust/alpine-allow-crt-static.patch]: Yes, I'm sure. It works on stable from this version

File: community/rust/APKBUILD (hunk 1 above)
Comment (jirutka) [reply community/rust/APKBUILD]: What is the error? If it doesn’t build without it, then how you can be sure it’s not really used?

File: community/rust/alpine-allow-crt-static.patch (hunk 2 above)
Comment (jirutka) [reply community/rust/alpine-allow-crt-static.patch]: Interesting. Even on official build from Rust team?

File: community/rust/musl-support-dynamic-linking.patch (hunk 3 above)
Comment (jeizsm) [reply community/rust/musl-support-dynamic-linking.patch]: I just copy it from file to fix patch

File: community/rust/alpine-allow-crt-static.patch (hunk 2 above)
Comment (jeizsm) [reply community/rust/alpine-allow-crt-static.patch]: Yes, I try it on version from rustup

File: community/rust/musl-support-dynamic-linking.patch (hunk 3 above)
Comment (jirutka) [reply community/rust/musl-support-dynamic-linking.patch]: Eh, yeah, I misunderstood it. :)

File: community/rust/APKBUILD (hunk 1 above)
Comment (jeizsm) [reply community/rust/APKBUILD]: I don't save error, but it something about missing files. I compile hello world with builded version and stripped binary about 150kb size and I don't see jemalloc string in binary
